runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -227,7 +230,20 @@ public void write(IToken token, Object data) throws OverwriteException {
      */
     public @Nonnull ILogData read(long address) {
         if (!runtime.getParameters().isCacheDisabled()) {
-            ILogData data = readCache.get(address);
+            ILogData data;
+            try {
+                data = readCache.get(address);
+            } catch (ExecutionException | UncheckedExecutionException e) {
+                // Guava wraps the exceptions thrown from the lower layers, therefore
+                // we need to unwrap them before throwing them to the upper layers that
+                // don't understand the guava exceptions
+                Throwable cause = e.getCause();
+                if (cause instanceof RuntimeException) {
+                    throw (RuntimeException) cause;
+                } else {
+                    throw new RuntimeException(cause);
```

Comment:
Please don't throw RuntimeException it's too generic, you can use at least IllegalStateException, the best case could be something like AddressSpaceViewException or ViewException

Reply:
If you have Guava then you can use `Throwables`:

```java
Throwables.throwIfUnchecked(t.getCause());
throw new IllegalStateException(t.getCause());
```

Reply:
So, since we try to get access to the data via guava, let's create and throw an `org.corfudb.runtime.exceptions.LogDataException extends RuntimeException`